cmake/Modules/FindOpenBLAS.cmake

```diff
@@ -14,78 +14,168 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+#
+# Finds the OpenBLAS library.
+#
+# The following variables are set after configuration is done:
+#
+# - OpenBLAS_FOUND
+# - OpenBLAS_INCLUDE_DIRS
+# - OpenBLAS_LIBRARIES
+#
+# This script will try to find the OpenBLAS library using the following mechanisms in this order:
+#
+# 1 - Use find_package(OpenBLAS) in Config mode. This is the recommended way to find dependencies
+#     with cmake.
+# 2 - Use pkg-config
+# 3 - Find the files manually
+#
+# At each step that was just described, in order to guarantee that the library was found correctly,
+# this script tries to compile this simple program:
+#
+# #include <cblas.h>
+# void main() { cblas_sasum(0, (float*)0, 0); }
+#
+# If this simple program can't be compiled with the found include directories provided by the step,
```

Comment:
How about this - 
"If this simple program can't be compiled with the OpenBLAS library discovered in the previous step,"
does this sound simpler?

Reply:
This looks better, I will change the text. Thanks for the suggestion.